components/ai_chat/content/browser/associated_web_contents_content.h

```diff
@@ -0,0 +1,188 @@
+
+  void SetPageContentFetcherDelegateForTesting(
+      std::unique_ptr<PageContentFetcherDelegate> delegate) {
+    page_content_fetcher_delegate_ = std::move(delegate);
+  }
+  raw_ptr<PageContentFetcherDelegate>
+  GetPageContentFetcherDelegateForTesting() {
+    return page_content_fetcher_delegate_.get();
+  }
+  raw_ptr<PrintPreviewExtractionDelegate>
+  GetPrintPreviewExtractionDelegateForTesting() {
+    return print_preview_extraction_delegate_.get();
+  }
+
+  void GetOpenAIChatButtonNonce(
+      mojom::PageContentExtractor::GetOpenAIChatButtonNonceCallback callback);
+
+  base::WeakPtr<AssociatedWebContentsContent> GetWeakPtr() {
+    return weak_ptr_factory_.GetWeakPtr();
+  }
+
+ private:
+  friend class ::AIChatUIBrowserTest;
+  friend class AssociatedWebContentsContentUnitTest;
+
+  // content::WebContentsObserver
+  void NavigationEntryCommitted(
+      const content::LoadCommittedDetails& load_details) override;
+  void TitleWasSet(content::NavigationEntry* entry) override;
+  void DidFinishLoad(content::RenderFrameHost* render_frame_host,
+                     const GURL& validated_url) override;
+
+  // ai_chat::AssociatedContentDriver
+  void GetPageContent(FetchPageContentCallback callback,
+                      std::string_view invalidation_token) override;
+  void OnNewPage(int64_t navigation_id) override;
+
+  // Called when an event of significance occurs that, if the page is a
+  // same-document navigation, should result in that previous navigation
+  // being considered as a new page.
+  void MaybeSameDocumentIsNewPage();
+
+  void GetSearchSummarizerKey(GetSearchSummarizerKeyCallback callback) override;
+
+  bool HasOpenAIChatPermission() const override;
+
+  void GetScreenshots(
+      mojom::ConversationHandler::GetScreenshotsCallback callback) override;
+
+  void OnScreenshotsCaptured(
+      mojom::ConversationHandler::GetScreenshotsCallback callback,
+      base::expected<std::vector<std::vector<uint8_t>>, std::string>);
+
+  void OnFetchPageContentComplete(FetchPageContentCallback callback,
+                                  std::string content,
+                                  bool is_video,
+                                  std::string invalidation_token);
+
+#if BUILDFLAG(ENABLE_PDF)
+  void OnPDFDocumentLoadComplete(FetchPageContentCallback callback);
+
+  void OnGetPDFPageCount(FetchPageContentCallback callback,
+                         pdf::mojom::PdfListener::GetPdfBytesStatus status,
+                         const std::vector<uint8_t>& bytes,
+                         uint32_t page_count);
+
+  void OnAllPDFPagesTextReceived(
+      FetchPageContentCallback callback,
+      const std::vector<std::pair<size_t, std::string>>& page_texts);
+#endif  // BUILDFLAG(ENABLE_PDF)
+
+  void SetPendingGetContentCallback(FetchPageContentCallback callback);
+
+  raw_ptr<AIChatMetrics> ai_chat_metrics_;
+
+  bool is_same_document_navigation_ = false;
+  int pending_navigation_id_ = -1;
```

Comment:
I feel we need some context now why `-1` makes sense in the comments...

Reply:
pushed

Reply:
https://github.com/brave/brave-core/pull/31797/commits/57b556ae06da93dde05b6c49d5394da89f420c2f